### PR TITLE
Fix handling results from metadatastore

### PR DESCRIPTION
**Pull request by jellonek**
Fix issue mentioned in #621 and rest of similar found in code.

---
This change is [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/622)

### CODE REVIEW

File: pkg/stream/streamer.go [hunk 1]
```diff
@@ -142,9 +142,12 @@ func (s *Server) getPodSandboxIP(sandboxID string) (string, error) {
 	sandbox := s.metadataStore.PodSandbox(sandboxID)
 	sandboxInfo, err := sandbox.Retrieve()
 	if err != nil {
-		glog.Errorf("Error when getting pod sandbox '%s': %v", sandboxID, err)
+		glog.Errorf("Error when getting pod sandbox %q: %v", sandboxID, err)
 		return "", err
 	}
+	if sandboxInfo == nil {
+		return "", fmt.Errorf("missing metadata for pod sandbox %q", sandboxID)
```
Comment (yanxuean) [inline pkg/stream/streamer.go]: add log

**Comment by yanxuean** [comment]
a nit,   
/lgtm

**Comment by yanxuean** [comment]
There are two lack.   
pkg/metadata/sandboxes_test.go   and  pkg/tools/dumpmetadata.go

**Comment by jellonek** [comment]
In dumpmetadata it's not needed (`spew` does not have any problem with dumping `nil` value) but in sandboxes test you have right. I'm adding that.

**Comment by jellonek** [comment]
Done.

**Comment by ivan4th** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/manager/manager.go, line 319 at r2](https://beta.reviewable.io/reviews/mirantis/virtlet/622#-L7UO4fm2Osf3PLii73L:-L7UNx0aB35GKA2x4zFs:bxc6cez) ([raw file](https://github.com/mirantis/virtlet/blob/0fec711f845433afbe4deac54c1378bc8568785c/pkg/manager/manager.go#L319)):*
> ```Go
> 	if sandboxInfo == nil {
> 		glog.Errorf("Sandbox %q doesn't exist", podSandboxID)
> 		return nil, fmt.Errorf("sandbox %q not found by Virtlet", podSandboxID)
> ```

IMO it would be better to say "in Virtlet metadata db" than "by Virtlet" (the latter doesn't explain where Virtlet was looking for it)

---

*[pkg/manager/manager.go, line 400 at r2](https://beta.reviewable.io/reviews/mirantis/virtlet/622#-L7UOHkP1vULdDVm_uR4:-L7UOHkP1vULdDVm_uR5:b2nreh6) ([raw file](https://github.com/mirantis/virtlet/blob/0fec711f845433afbe4deac54c1378bc8568785c/pkg/manager/manager.go#L400)):*
> ```Go
> 	if sandboxInfo == nil {
> 		glog.Errorf("Sandbox %q doesn't exist", podSandboxID)
> 		return nil, fmt.Errorf("sandbox %q not found by Virtlet", podSandboxID)
> ```

Same as above.

---

*[pkg/metadata/sandboxes_test.go, line 106 at r2](https://beta.reviewable.io/reviews/mirantis/virtlet/622#-L7UOQst7-fMIfn2HfWw:-L7UOQst7-fMIfn2HfWx:b-etu6k1) ([raw file](https://github.com/mirantis/virtlet/blob/0fec711f845433afbe4deac54c1378bc8568785c/pkg/metadata/sandboxes_test.go#L106)):*
> ```Go
> 		if actualSandboxInfo == nil {
> 			t.Fatal(fmt.Errorf("missing PodSandboxInfo for sandbox %q", sandbox.GetMetadata().Uid))
> 		}
> ```

Can we also add a simple test case that verifies that nil is returned when the sandbox is not found?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mirantis/virtlet/622)*

**Comment by jellonek** [comment]
Review status: 0 of 5 files reviewed at latest revision, 4 unresolved discussions.

---

*[pkg/manager/manager.go, line 319 at r2](https://beta.reviewable.io/reviews/mirantis/virtlet/622#-L7UO4fm2Osf3PLii73L:-L7UR86ICvhD80u9Y1br:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/0fec711f845433afbe4deac54c1378bc8568785c/pkg/manager/manager.go#L319)):*
<details><summary><i>Previously, ivan4th (Ivan Shvedunov) wrote…</i></summary><blockquote>

IMO it would be better to say "in Virtlet metadata db" than "by Virtlet" (the latter doesn't explain where Virtlet was looking for it)
</blockquote></details>

Done.

---

*[pkg/manager/manager.go, line 400 at r2](https://beta.reviewable.io/reviews/mirantis/virtlet/622#-L7UOHkP1vULdDVm_uR4:-L7UR8ThD1Z8bAuGxyT0:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/0fec711f845433afbe4deac54c1378bc8568785c/pkg/manager/manager.go#L400)):*
<details><summary><i>Previously, ivan4th (Ivan Shvedunov) wrote…</i></summary><blockquote>

Same as above.
</blockquote></details>

Done.

---

*[pkg/metadata/sandboxes_test.go, line 106 at r2](https://beta.reviewable.io/reviews/mirantis/virtlet/622#-L7UOQst7-fMIfn2HfWw:-L7UYbzG8XqHRBZYEzN8:b-ehxek9) ([raw file](https://github.com/mirantis/virtlet/blob/0fec711f845433afbe4deac54c1378bc8568785c/pkg/metadata/sandboxes_test.go#L106)):*
<details><summary><i>Previously, ivan4th (Ivan Shvedunov) wrote…</i></summary><blockquote>

Can we also add a simple test case that verifies that nil is returned when the sandbox is not found?
</blockquote></details>

It's not so simple because that can occur only when we have a created bucket for sandbox and in the same time it's empty. That's not normal situation and can not be achieved using our abstraction, as we don't have a separate func for bucket creation and saving data to it, we are doing that in single transaction which when failed - will roll back creation of empty bucket.

So i wonder how the issue mentioned in #621 occurred and how to replicate it.
Creation of empty sandbox is blocked by https://github.com/Mirantis/virtlet/blob/master/pkg/metadata/sandboxes.go#L136 

---

*[pkg/stream/streamer.go, line 149 at r1](https://beta.reviewable.io/reviews/mirantis/virtlet/622#-L7U9tJG_C_4kq7k-i3p-r1-149:-L7US3TH1cQE0LapwokD:b-896fix) ([raw file](https://github.com/mirantis/virtlet/blob/924b3bfd2c0dc8c2f42be1a9d0779d2f820e78f8/pkg/stream/streamer.go#L149)):*
<details><summary><i>Previously, yanxuean wrote…</i></summary><blockquote>

add log
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mirantis/virtlet/622)*

**Comment by ivan4th** [comment]
:lgtm:

---

Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/622#-:-L81LHZjTaPv5UypLSlX:bnfp4nl)*

**Comment by lukaszo** [comment]
:lgtm:

---

Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/622#-:-L8CMyW66oJa_yyn3TW8:bnfp4nl)*
